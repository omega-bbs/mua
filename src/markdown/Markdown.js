import React from 'react'
import PropTypes from 'prop-types'
import { Parser } from 'commonmark'
import ReactRenderer from 'commonmark-react-renderer'

const parser = new Parser()
const renderer = new ReactRenderer()

class Markdown extends React.Component {
  static propTypes = {
    markdown: PropTypes.string.isRequired,
  }

  state = {
    ast: null,
    vdom: null,
  }

  componentWillMount() {
    this.parse(this.props.markdown)
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.markdown !== this.props.markdown) {
      this.parse(nextProps.markdown)
    }
  }

  parse(markdown) {
    const ast = parser.parse(markdown)
    const vdom = renderer.render(ast)
    this.setState({
      ast,
      vdom,
    })
  }

  render() {
    return (
      <div>
        {this.state.vdom}
      </div>
    )
  }
}

export default Markdown
