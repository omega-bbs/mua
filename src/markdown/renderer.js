import ReactRenderer from 'commonmark-react-renderer'

const createRenderer = () => {
  return new ReactRenderer({
    skipHtml: true,
    softBreak: 'br',
  })
}

export default {
  renderer: null,

  render(ast) {
    if (!this.renderer) {
      this.renderer = createRenderer()
    }
    return this.renderer.render(ast)
  },
}
