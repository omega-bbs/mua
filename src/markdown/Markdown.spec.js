import React from 'react'
import renderer from 'react-test-renderer'

import Markdown from './Markdown'

const renderMarkdown = markdown => {
  return renderer.create(<Markdown markdown={markdown} />).toJSON()
}

test('simple', () => {
  expect(renderMarkdown('*italic* **bold** `code`')).toMatchSnapshot()
})
