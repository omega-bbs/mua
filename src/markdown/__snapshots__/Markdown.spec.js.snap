// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple 1`] = `
<div>
  <p>
    <em>
      italic
    </em>
     
    <strong>
      bold
    </strong>
     
    <code
      className={undefined}
      data-sourcepos={undefined}
    >
      code
    </code>
  </p>
</div>
`;
